slicedimage/cli/checksum.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import hashlib
+import os
+import sys
+
+from slicedimage import Reader, Writer
+from ._base import CliCommand
+
+
+class ChecksumCommand(CliCommand):
+    @classmethod
+    def register_parser(cls, subparser_root):
+        checksum_command = subparser_root.add_parser("checksum", help="Read a TOC file and add missing checksums.")
+        checksum_command.add_argument("in_url", help="URL for the source TOC file")
+        checksum_command.add_argument("out_path", help="Path to write TOC file with checksums")
+        checksum_command.add_argument("--pretty", action="store_true", help="Pretty-print the output file")
+
+        return checksum_command
+
+    @classmethod
+    def run_command(cls, args):
+        try:
+            slicedimage = Reader.parse_doc(args.in_url, None)
+        except ValueError:
+            if os.path.exists(args.in_url):
```

Comment:
should this be `isfile` instead of `exists`?

Reply:
will fix.